### PR TITLE
Add top visited pages for last week table

### PR DESCRIPTION
**Pull request by jonathanloos**
- Display the top 10 controller#controller_action combinations based on the number of unique users that have triggered that action
- Show total number of actions unique users have triggered

### CODE REVIEW

File: app/components/matey/top_visited_pages_table_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<div class="card w-50 h-25">
+  <div class="card-body">
+    <h1 class="card-title">
+      Top 10 Page Visits In the Last Week
```
Comment (krsyoung) [inline app/components/matey/top_visited_pages_table_component.html.erb:4]: Thoughts on hardcoding this if the `time_window` could be variable?

**Comment by suvasan19** [comment]
I was trying to add the component into the sample app to see how it looks, but after some playing around I realized we can't use Ahoy::Events to display these events since data like "action" within properties is unique based on the configurations of the user application.

For example, in Matey, we only track 1 type of event currently, and they all look the same with no specific visit data.

image

Thankfully, we can use the **Ahoy::Visit** table, which keeps track of the pages for all visits!

`Ahoy::Visit.where(started_at: 1.week.ago..Time.zone.today).pluck(:landing_page).tally` will return a hash with the full path and number of visits

**Comment by krsyoung** [comment]
We could also make the parameter configurable? Default to something reasonable like `action` and then let the user override it?

**Comment by suvasan19** [comment]
Yup, but the Ahoy::Visits data will give us the info we need and is standard across any app that will use Ahoy, I wouldn't mind being able to pass in the data that the user wants to show, so having the default as visits and allowing for configurations to the data source

**Comment by jonathanloos** [comment]
@suvasan19 a visit will contain references to many events, which makes me think it's too much of a scope out. All of the pages the user hits are tracked in events and linked to a particular visit. The landing page being only the first page the user lands on, but we would lose the data on the rest of their visit. I would be more inclined to go with @krsyoung's suggestion of passing in a property value so that it can be customized on a per-app basis. Thoughts?

**Comment by suvasan19** [comment]
Ya you're right, User's should have some events set up anyway. Let's get that in and we can see how it looks on the sample app! Nice additions!

**Pull request merged**